### PR TITLE
#17, #116, #136, #150, #153, #193 - fixes from Valera

### PR DESCRIPTION
**Pull request by MickeyMouse2**

### CODE REVIEW

**Comment by kirillt** [comment]
Build: https://www.taran.space/dev/ark-navigator-valera.apk

File: app/src/main/java/space/taran/arknavigator/mvp/model/repo/nodes/FolderNodesRepo.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import space.taran.arknavigator.mvp.presenter.adapter.folderstree.FolderNode
+
+class FolderNodesRepo(
+    private val nodesProvider: FolderNodesProvider
+) {
+    fun save(nodes: List<FolderNode>) = nodesProvider.save(nodes)
+
+    fun provide() = nodesProvider.provide()
+}
```
Comment (mdrlzy) [inline app/src/main/java/space/taran/arknavigator/mvp/model/repo/nodes/FolderNodesRepo.kt:11]: As far as I understand the main idea is to have an application level cache for folder tree items. Not sure we need this.

File: app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/PreviewsPagerPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import space.taran.arknavigator.mvp.model.repo.index.ResourceMeta
+import space.taran.arknavigator.mvp.model.repo.index.ResourcesIndex
+import space.taran.arknavigator.mvp.model.repo.preview.PreviewAndThumbnail
+import space.taran.arknavigator.mvp.view.GalleryView
+import space.taran.arknavigator.mvp.view.item.PreviewItemView
+import space.taran.arknavigator.utils.ImageUtils
+import space.taran.arknavigator.utils.extension
+
+class PreviewsPagerPresenter(val viewState: GalleryView) {
```
Comment (mdrlzy) [inline app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/PreviewsPagerPresenter.kt:11]: Outdated file, it was removed in the main branch.

File: app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/TextFileOpener.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package space.taran.arknavigator.mvp.presenter.adapter
+
+import moxy.viewstate.strategy.alias.AddToEndSingle
+
+interface TextFileOpener {
```
Comment (mdrlzy) [inline app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/TextFileOpener.kt:5]: What is this interface for?

**Comment by cnzqa** [comment]
#17 still active
#116 issue active, there is no preview and edit crash app
``` 2022-04-01 20:41:11.812 28207-28207/space.taran.arknavigator E/AndroidRuntime: FATAL EXCEPTION: main
    Process: space.taran.arknavigator, PID: 28207
    android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.EDIT dat=content://space.taran.arknavigator.provider/external/Download/test.txt typ=text/plain flg=0x3 (has extras) }```
 #136  is ok
 #150 is ok
 #153 not checeked, need hint here
 #193 Is ok

**Pull request closed without merging**